src/main/java/net/minecraftforge/common/data/ForgeEntityTypeTagsProvider.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2021.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common.data;
+
+import net.minecraft.data.DataGenerator;
+import net.minecraft.data.tags.EntityTypeTagsProvider;
+import net.minecraft.world.entity.EntityType;
+import net.minecraftforge.common.Tags;
+
+public class ForgeEntityTypeTagsProvider extends EntityTypeTagsProvider {
+
+    public ForgeEntityTypeTagsProvider(DataGenerator generator, String modId, ExistingFileHelper existingFileHelper)
```

Comment:
The call initializing and adding this provider appears to be missing.

Reply:
No, if you actually try running the datagen you will see the file you manually created in the generated folder get deleted... Providers forge needs to run should be added in forge's [GatherDataEvent listener](https://github.com/MinecraftForge/MinecraftForge/blob/1.18.x/src/main/java/net/minecraftforge/common/ForgeMod.java#L209).

Reply:
I'll add that in my next commit and push.